<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Benchmarking clustifyr • clustifyrdata</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Benchmarking clustifyr">
<meta property="og:description" content="clustifyrdata">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">clustifyrdata</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/benchmark.html">Benchmarking clustifyr</a>
    </li>
    <li>
      <a href="../articles/download_refs.html">Prebuilt references for clustifyr</a>
    </li>
    <li>
      <a href="../articles/geo-annotations.html">Improving NCBI GEO submissions of scRNA-seq data</a>
    </li>
    <li>
      <a href="../articles/otherformats.html">Support for single cell object formats</a>
    </li>
    <li>
      <a href="../articles/othertools.html">Other considerations w/ clustifyr</a>
    </li>
    <li>
      <a href="../articles/references.html">Reference building for clustifyr</a>
    </li>
    <li>
      <a href="../articles/visualizations.html">Visualization of clustifyr results</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/rnabioco/clustifyrdata/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="benchmark_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Benchmarking clustifyr</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/rnabioco/clustifyrdata/blob/master/vignettes/benchmark.Rmd"><code>vignettes/benchmark.Rmd</code></a></small>
      <div class="hidden name"><code>benchmark.Rmd</code></div>

    </div>

    
    
<div id="benchmark-1--mca-lung-dataset-annotation-using-ref_tabula_muris_drop-reference" class="section level2">
<h2 class="hasAnchor">
<a href="#benchmark-1--mca-lung-dataset-annotation-using-ref_tabula_muris_drop-reference" class="anchor"></a>Benchmark 1. MCA lung dataset annotation using <code>ref_tabula_muris_drop</code> reference</h2>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://github.com/rnabioco/clustifyr#readme">clustifyr</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://rnabioco.github.io/clustifyrdata">clustifyrdata</a></span>)

<span class="kw">l_mat</span> <span class="op">&lt;-</span> <span class="kw">clustifyrdata</span>::<span class="kw"><a href="../reference/MCA_lung_mat.html">MCA_lung_mat</a></span>
<span class="kw">l_meta</span> <span class="op">&lt;-</span> <span class="kw">clustifyrdata</span>::<span class="kw"><a href="../reference/MCA_lung_meta.html">MCA_lung_meta</a></span>

<span class="co"># find lung references, remove generic terms</span>
<span class="kw">lung_cols</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"-Lung"</span>,
                   <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="kw">ref_tabula_muris_drop</span>), 
                   value = <span class="fl">TRUE</span>)

<span class="kw">tml_ref</span> <span class="op">&lt;-</span> <span class="kw">ref_tabula_muris_drop</span>[, <span class="kw">lung_cols</span>]
<span class="kw">tml_ref</span> <span class="op">&lt;-</span> <span class="kw">tml_ref</span>[, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">8</span>, <span class="fl">13</span>)]

<span class="co"># default with all genes</span>
<span class="kw">start</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/proc.time.html">proc.time</a></span>()

<span class="kw">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/clustify.html">clustify</a></span>(
  input = <span class="kw">l_mat</span>,
  ref_mat = <span class="kw">tml_ref</span>,
  metadata = <span class="kw">l_meta</span>,
  cluster_col = <span class="st">"Annotation"</span>
)

<span class="kw">res_allgenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/cor_to_call.html">cor_to_call</a></span>(
  cor_mat = <span class="kw">res</span>,
  metadata = <span class="kw">l_meta</span>,
  cluster_col = <span class="st">"Annotation"</span>
)

<span class="kw">end</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/proc.time.html">proc.time</a></span>()

<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">res_allgenes</span>) <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"MCA annotation"</span>, <span class="st">"clustifyr call"</span>, <span class="st">"r"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">end</span> <span class="op">-</span> <span class="kw">start</span>)
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;   2.109   0.469   2.578</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">res_allgenes</span>, n = <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="kw">res_allgenes</span>))
<span class="co">#&gt; # A tibble: 32 x 3</span>
<span class="co">#&gt; # Groups:   MCA annotation [32]</span>
<span class="co">#&gt;    `MCA annotation`                `clustifyr call`                            r</span>
<span class="co">#&gt;    &lt;chr&gt;                           &lt;chr&gt;                                   &lt;dbl&gt;</span>
<span class="co">#&gt;  1 Alveolar macrophage_Ear2 high(… alveolar macrophage-Lung                0.878</span>
<span class="co">#&gt;  2 Alveolar macrophage_Pclaf high… alveolar macrophage-Lung                0.714</span>
<span class="co">#&gt;  3 B Cell(Lung)                    B cell-Lung                             0.836</span>
<span class="co">#&gt;  4 Ig−producing B cell(Lung)       B cell-Lung                             0.577</span>
<span class="co">#&gt;  5 Ciliated cell(Lung)             ciliated columnar cell of tracheobronc… 0.820</span>
<span class="co">#&gt;  6 Plasmacytoid dendritic cell(Lu… classical monocyte-Lung-CLASH!          0.847</span>
<span class="co">#&gt;  7 Eosinophil granulocyte(Lung)    leukocyte-Lung                          0.716</span>
<span class="co">#&gt;  8 Neutrophil granulocyte(Lung)    leukocyte-Lung                          0.634</span>
<span class="co">#&gt;  9 Endothelial cell_Kdr high(Lung) lung endothelial cell-Lung              0.747</span>
<span class="co">#&gt; 10 Endothelial cell_Tmem100 high(… lung endothelial cell-Lung              0.803</span>
<span class="co">#&gt; 11 Endothelial cells_Vwf high(Lun… lung endothelial cell-Lung              0.764</span>
<span class="co">#&gt; 12 Basophil(Lung)                  mast cell-Lung                          0.440</span>
<span class="co">#&gt; 13 NK Cell(Lung)                   natural killer cell-Lung                0.804</span>
<span class="co">#&gt; 14 Conventional dendritic cell_Gn… non-classical monocyte-Lung-CLASH!      0.789</span>
<span class="co">#&gt; 15 Stromal cell_Acta2 high(Lung)   stromal cell-Lung                       0.646</span>
<span class="co">#&gt; 16 Stromal cell_Dcn high(Lung)     stromal cell-Lung                       0.814</span>
<span class="co">#&gt; 17 Stromal cell_Inmt high(Lung)    stromal cell-Lung                       0.817</span>
<span class="co">#&gt; 18 Dividing T cells(Lung)          T cell-Lung                             0.720</span>
<span class="co">#&gt; 19 Nuocyte(Lung)                   T cell-Lung                             0.758</span>
<span class="co">#&gt; 20 T Cell_Cd8b1 high(Lung)         T cell-Lung                             0.826</span>
<span class="co">#&gt; 21 Alveolar bipotent progenitor(L… alveolar epithelial type 2 cells-Lung   0.663</span>
<span class="co">#&gt; 22 AT1 Cell(Lung)                  alveolar epithelial type 2 cells-Lung   0.770</span>
<span class="co">#&gt; 23 AT2 Cell(Lung)                  alveolar epithelial type 2 cells-Lung   0.880</span>
<span class="co">#&gt; 24 Clara Cell(Lung)                alveolar epithelial type 2 cells-Lung   0.733</span>
<span class="co">#&gt; 25 Dividing cells(Lung)            alveolar epithelial type 2 cells-Lung   0.647</span>
<span class="co">#&gt; 26 Conventional dendritic cell_H2… dendritic cells and interstital macrop… 0.550</span>
<span class="co">#&gt; 27 Conventional dendritic cell_Mg… dendritic cells and interstital macrop… 0.788</span>
<span class="co">#&gt; 28 Conventional dendritic cell_Tu… dendritic cells and interstital macrop… 0.671</span>
<span class="co">#&gt; 29 Dendritic cell_Naaa high(Lung)  dendritic cells and interstital macrop… 0.802</span>
<span class="co">#&gt; 30 Dividing dendritic cells(Lung)  dendritic cells and interstital macrop… 0.676</span>
<span class="co">#&gt; 31 Interstitial macrophage(Lung)   dendritic cells and interstital macrop… 0.804</span>
<span class="co">#&gt; 32 Monocyte progenitor cell(Lung)  dendritic cells and interstital macrop… 0.581</span>
</pre></div>
</div>
<div id="benchmark-2--using-sorted-microarray-data-to-classify-10x-pbmc-example-data-available-in-clustifyrdata-package" class="section level2">
<h2 class="hasAnchor">
<a href="#benchmark-2--using-sorted-microarray-data-to-classify-10x-pbmc-example-data-available-in-clustifyrdata-package" class="anchor"></a>benchmark 2. Using sorted microarray data to classify 10x PBMC example data, available in <code>clustifyrdata</code> package</h2>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw">full_pbmc_matrix</span> <span class="op">&lt;-</span> <span class="kw">clustifyrdata</span>::<span class="kw"><a href="../reference/pbmc_matrix.html">pbmc_matrix</a></span>
<span class="kw">full_pbmc_meta</span> <span class="op">&lt;-</span> <span class="kw">clustifyrdata</span>::<span class="kw"><a href="../reference/pbmc_meta.html">pbmc_meta</a></span>
<span class="kw">microarray_ref</span> <span class="op">&lt;-</span> <span class="kw">clustifyrdata</span>::<span class="kw"><a href="../reference/ref_hema_microarray.html">ref_hema_microarray</a></span>

<span class="kw">start</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/proc.time.html">proc.time</a></span>()

<span class="kw">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/clustify.html">clustify</a></span>(
  input = <span class="kw">full_pbmc_matrix</span>,
  ref_mat = <span class="kw">microarray_ref</span>,
  metadata = <span class="kw">full_pbmc_meta</span>,
  query_genes = <span class="kw">pbmc_vargenes</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">500</span>],
  cluster_col = <span class="st">"classified"</span>
)

<span class="kw">res2</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/cor_to_call.html">cor_to_call</a></span>(<span class="kw">res</span>, threshold = <span class="fl">0.5</span>)

<span class="kw">end</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/proc.time.html">proc.time</a></span>()

<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">res2</span>) <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"manual annotation"</span>, <span class="st">"clustifyr call"</span>, <span class="st">"r"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">end</span> <span class="op">-</span> <span class="kw">start</span>)
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;   0.076   0.005   0.081</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">res2</span>, n = <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="kw">res2</span>))
<span class="co">#&gt; # A tibble: 9 x 3</span>
<span class="co">#&gt; # Groups:   manual annotation [9]</span>
<span class="co">#&gt;   `manual annotation` `clustifyr call`                    r</span>
<span class="co">#&gt;   &lt;chr&gt;               &lt;chr&gt;                           &lt;dbl&gt;</span>
<span class="co">#&gt; 1 Memory CD4 T        CD4+ Effector Memory            0.585</span>
<span class="co">#&gt; 2 Naive CD4 T         CD4+ Effector Memory            0.594</span>
<span class="co">#&gt; 3 CD8 T               CD8+ Effector Memory            0.602</span>
<span class="co">#&gt; 4 NK                  Mature NK cell_CD56+ CD16+ CD3- 0.537</span>
<span class="co">#&gt; 5 Platelet            unassigned                      0.298</span>
<span class="co">#&gt; 6 CD14+ Mono          Monocyte                        0.593</span>
<span class="co">#&gt; 7 FCGR3A+ Mono        Monocyte                        0.559</span>
<span class="co">#&gt; 8 DC                  Myeloid Dendritic Cell          0.556</span>
<span class="co">#&gt; 9 B                   Naïve B-cells                   0.634</span>
</pre></div>
<ol style="list-style-type: decimal">
<li>Please see <a href="https://f1000research.com/articles/9-223">manuscript</a> for full benchmarking.</li>
</ol>
</div>
<div id="comparison-with-other-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#comparison-with-other-methods" class="anchor"></a>Comparison with other methods</h2>
<p>using Tablua Muris (drop and facs samples) 12 shared tissues, which can be downloaded as <code>seurat</code> objects</p>
<ol start="2" style="list-style-type: decimal">
<li>Building reference and then mapping:</li>
</ol>
<p>default <code>clustify</code>, with all genes</p>
<p><code>clustify</code>, pulling <code>var.genes</code> from <code>seurat</code> objects</p>
<p><code>clustify</code>, using <code>M3Drop</code> for feature selection</p>
<p><code>clustify</code>, using <code>per_cell = TRUE</code> option, and then assign cluster consensus ident with <code>collapse_to_cluster = TRUE</code></p>
<p><code>clustify</code>, after <code>ALRA</code> imputation, using <code>per_cell = TRUE</code> option, and then assign cluster consensus ident with <code>collapse_to_cluster = TRUE</code></p>
<p><code>scmap-cluster</code></p>
<ol start="3" style="list-style-type: decimal">
<li>Mapping from prebuilt all-encompassing references to the drop samples:</li>
</ol>
<p><code>clustify</code>, using <code>ref_tabula_muris_facs</code></p>
<p><code>singleR</code>, using default built-in mouse references without fine tuning</p>
<ol start="4" style="list-style-type: decimal">
<li>Generate marker gene list (of 30 genes per reference identity), and then mapping</li>
</ol>
<p>default <code>clustify_list</code></p>
<p><img src="img/test.png" width="1094" style="display: block; margin: auto;"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Rui Fu, Jay Hesselberth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
